File: dongmin/week8/포켓몬.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashSet<Integer> deduplication = new HashSet<>();
        for(int num : nums)
            deduplication.add(num);
        return nums.length/2<deduplication.size()?nums.length/2:deduplication.size();
    }
}

File: yeongseong/8week/포켓몬.js
function solution(nums) {
    const dedupe = new Set(nums);
    const dedupecount = dedupe.size;
    const numcount = nums.length;
    
    if (numcount / 2 > dedupecount){
        return dedupecount;
    } else {
        return numcount / 2;
    }
}